test/tools/TestExe/TestExe.cs

```diff
@@ -52,6 +60,41 @@ private static void EchoArgs(string[] args)
             }
         }
 
+        [DllImport("Kernel32.dll")]
+        private static extern IntPtr GetCommandLineW();
+
+        // <Summary>
+        // Echos the raw command line received by the process plus the arguments passed in.
+        // </Summary>
+        private static void EchoRaw(string[] args)
+        {
+            string rawCmdLine = Environment.CommandLine;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                IntPtr cmdLinePtr = GetCommandLineW();
+                rawCmdLine = Marshal.PtrToStringUni(cmdLinePtr);
+            }
+
+            Console.WriteLine("Raw Command Line:\n{0}\n", rawCmdLine);
```

Comment:
As argued [here](https://github.com/PowerShell/PowerShell/issues/18554#issuecomment-1318664972), it's probably better simply not to show a "raw" command line on Unix, given that there is no such thing, and given that the _hypothetical shell command line_ that `Environment.CommandLine` reverse-engineers from the verbatim arguments doesn't even properly work in POSIX-compatible shells (no escaping of `\` and `$`). Thus, I suggest `string rawCmdLine = "N/A";`

Reply:
Updated to output `N/A` on non-windows.